tests/cpp/predictor/test_gpu_predictor.cu

```diff
@@ -76,5 +76,46 @@ TEST(gpu_predictor, Test) {
 
   delete dmat;
 }
+
+// multi-GPU predictor test
+TEST(gpu_predictor, MGPU_Test) {
+  std::unique_ptr<Predictor> gpu_predictor =
+      std::unique_ptr<Predictor>(Predictor::Create("gpu_predictor"));
+  std::unique_ptr<Predictor> cpu_predictor =
+      std::unique_ptr<Predictor>(Predictor::Create("cpu_predictor"));
+
+  gpu_predictor->Init({std::pair<std::string, std::string>("n_gpus", "-1")}, {});
+  cpu_predictor->Init({}, {});
+
+  std::vector<std::unique_ptr<RegTree>> trees;
+  trees.push_back(std::unique_ptr<RegTree>(new RegTree()));
+  trees.back()->InitModel();
+  (*trees.back())[0].SetLeaf(1.5f);
+  (*trees.back()).Stat(0).sum_hess = 1.0f;
+  gbm::GBTreeModel model(0.5);
+  model.CommitModel(std::move(trees), 0);
+  model.param.num_output_group = 1;
+
+  int n_row = 5;
```

Comment:
Changing this line to
```suggestion
  int n_row = 8;
```
gets rid of segmentation fault.

The p2.8xlarge instance has 8 GPUs, so with the matrix with 5 rows, some of the GPUs were getting 0 row. We should handle this edge case either by restricting the number of devices when too few rows are given, or by correctly handling zero-row shards.

Reply:
@trivialfis How should we handle this edge case?

Reply:
@hcho3 cuda-gdb doesn't work for me with NCCL, neither on Fedora nor on Ubuntu.. :(
The `GPUSet::All()` has an optional parameter specifying number of rows.

Reply:
Done.